python/foxglove-sdk/python/foxglove/notebook/foxglove_widget.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import importlib.metadata
+import pathlib
+from typing import TYPE_CHECKING, Any, Literal, TypedDict
+
+import anywidget
+import traitlets
+
+if TYPE_CHECKING:
+    from .notebook_buffer import NotebookBuffer
+
+try:
+    __version__ = importlib.metadata.version("foxglove")
+except importlib.metadata.PackageNotFoundError:
+    __version__ = "unknown"
+
+
+class SelectLayoutParams(TypedDict):
+    storageKey: str
+    opaqueLayout: dict
+    force: bool | None
+
+
+class FoxgloveWidget(anywidget.AnyWidget):
+    """
+    A widget that displays a Foxglove viewer in a notebook.
+
+    :param get_data: A callback function that returns the data to display in the widget.
+    :param width: The width of the widget. Defaults to "100%".
+    :param height: The height of the widget. Defaults to "500px".
+    :param src: The source URL of the Foxglove viewer. Defaults to "https://embed.foxglove.dev/".
+    :param layout_data: The layout data to use for the widget.
+    """
+
+    _esm = pathlib.Path(__file__).parent / "static" / "widget.js"
+    width = traitlets.Union([traitlets.Int(), traitlets.Enum(values=["full"])]).tag(
+        sync=True
+    )
+    height = traitlets.Int(500).tag(sync=True)
+    src = traitlets.Unicode("").tag(sync=True)
```

Comment:
is there a way to have the default be None instead of an empty string? It doesn't look like we have explicit defaults for some other values like `width`, yet we treat those ones as optional so I am wondering why we need a default value of `""` for this one?